app/jobs/regular/publish_group_membership_updates.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Jobs
+  class PublishGroupMembershipUpdates < ::Jobs::Base
+    def execute(args)
+      group = Group.find_by(id: args[:group_id])
+      users = User.human_users.where(id: args[:user_ids])
+
+      return if !group
+      return if !%w[add remove].include?(args[:type])
+
+      added_members = args[:type] == 'add'
+      event_name = added_members ? :user_added_to_group : :user_removed_from_group
+
+      users.each do |user|
+        params = [user, group]
+        params << { automatic: group.automatic? } if added_members
```

Comment:
Curious why this is necessary when we already pass the group object to the event. The listeners of the event can easily determine if the group is automatic or not.

Reply:
Agree, but we need to be consistent with the event triggered by `Group#add`

Reply:
O in this case, I think extracting the triggering of the event to a method in the group model will be better. Otherwise, `Group#add` may rename the event name in the future and this goes out of sync again.